R/finalize_plot.R

```diff
@@ -315,29 +338,113 @@ finalize_plot <- function(input_plot = NULL,
+                            )
+      )
+
+    # Extract the legend
+    l <- gtable::gtable_filter(gt, "guide-box")[[1]]
+
+    # Remove the legend from the plot
+    p <- p + ggplot2::theme(legend.position = "none")
+
+    # Create a buffer
+    buffer <- grid::rectGrob(gp = grid::gpar(col = "transparent", fill = "transparent"))
+
+    # Assemble
+    built <- gridExtra::grid.arrange(
+      gridExtra::arrangeGrob(
+        l[[1]],
+        buffer,
+        p,
+        nrow = 3,
+        heights = grid::unit(c(l_height,
+                               plot_constants$margin_v5,
+                               p_height),
+                             "bigpts")
+      )
+    )
+
+    return(built)
+  }
+
+  # Use helper function to develop full stack of legend, buffer, and plot
+  p_l_stack <- buildChart(processed_plot,
+                          l_margins = plot_constants$padding_legend,
+                          p_margins = plot_constants$padding_plot,
+                          l_indent = plot_constants$legend_indent,
+                          l_height = legend_height,
+                          p_height = plot_constants$height - legend_height - plot_constants$margin_v1_v2_v4_v5)
+
   # ggplot as grob (vp.plot)
   grob_plot <- grid::grobTree(
-    ggplotGrob(
-      input_plot + ggplot2::theme(
-        # make sure the plot has no title or caption
-        plot.title = ggplot2::element_blank(),
-        plot.caption = ggplot2::element_blank(),
-        # add margins
-        plot.margin = unit(plot_constants$padding_plot, "bigpts"),
-        # modify text sizing. **THIS MODIFICATION IS NEEDED BUT NOT UNDERSTOOD**
-        text = ggplot2::element_text(size = cmapplot_globals$font$main$size * 1.25)
-      )
-    ),
+    p_l_stack,
     vp = vp.plot,
     name = "plot"
   )
 
+  # If debug is called, assemble additional outlines for overlay
+  if (debug) {
+    debug_rects <- grid::rectGrob(gp = gpar(col = "black", fill = "transparent"))
+
+    debug_stack <- gridExtra::arrangeGrob(
+      debug_rects,
+      debug_rects,
+      debug_rects,
+      nrow = 3,
+      heights = grid::unit(c(legend_height,
+                             plot_constants$margin_v5,
+                             plot_constants$height - legend_height - plot_constants$margin_v1_v2_v4_v5),
+                           "bigpts")
+
+    )
+
+    debug_plot <- grid::grobTree(
+      debug_stack,
+      vp = vp.plot,
+      name = "debug_plot"
+    )
+  }
+
+
   # Assemble final plot -----------------------------------------------------
 
+  # If not debugging, proceed as normal
+  if(!debug) {
   final_plot <- grid::grobTree(
     grob_background, grob_topline, grob_title, grob_caption, grob_plot,
     name = "final_plot"
-  )
+    )
+  } else {
+    # If debugging, add outlines to text boxes and call debug outlines for plot
+    final_plot <- grid::grobTree(
+      grob_background, grob_topline,
+      grid::editGrob(grob_title, box_gp = grid::gpar(col = "black")),
```

Comment:
Would a preferred way of doing this be to do something like `if(debug){ grob_outline_color = "black" } else { grob_outline_color = "transparent" }` way up top, and then adding `box_gp = gpar(col =  grob_outline_color)` in all textbox_grobs in all situations?

Reply:
Makes sense.

Reply:
Or, better, maybe `box_gp = gpar(col = if(debug){"tomato1"}else{"transparent"})` where each grob gets a different color outline (yes, tomato1 is a [real color](http://sape.inf.usi.ch/quick-reference/ggplot2/colour)). 

I'd experiment with this first to make sure `gpar(col = "transparent"}` works... maybe it prefers NULL?

Reply:
Incorporated.

Reply:
Also - `transparent` does work. I tried out a few things.

I'm going to leave it as all black for now, different colors for each would make the code much more complex (and I like the simplicity of the new approach).